src/script/interpreter.cpp

```diff
@@ -1279,18 +1279,29 @@ uint256 GetOutputsHash(const T& txTo)
 } // namespace
 
 template <class T>
-PrecomputedTransactionData::PrecomputedTransactionData(const T& txTo)
+void PrecomputedTransactionData::Init(const T& txTo)
 {
+    assert(!m_ready);
+
     // Cache is calculated only for transactions with witness
     if (txTo.HasWitness()) {
         hashPrevouts = GetPrevoutHash(txTo);
         hashSequence = GetSequenceHash(txTo);
         hashOutputs = GetOutputsHash(txTo);
-        ready = true;
     }
+
+    m_ready = true;
```

Comment:
Why move `m_ready = true;` outside of the `if`? This makes it possible to end up in a weird state where `m_ready` is `true`, but the member variables are not initialized (if `txTo.HasWitness()` is `false`).

Reply:
Because `m_ready` now means 'is this object initialized' (see conversation at https://github.com/bitcoin/bitcoin/pull/18401#discussion_r401812158). A future commit in #17977 changes `PrecomputedTransactionData` to store data even for non-segwit transactions (https://github.com/bitcoin/bitcoin/pull/17977/commits/6dcc85e3347fe8a0c5e3e578176fd38fa093df39), so `m_ready` would need to be set outside the `if (txTo.HasWitness)` block.